src/Instances/New.lua

```diff
@@ -71,12 +71,22 @@ local function New(className: string)
 
 				-- Properties bound to state
 				if typeof(value) == "table" and value.type == "State" then
-					local assignOK = pcall(function()
-						ref.instance[key] = value:get(false)
+					local existingValue, attemptedValue
+					local assignOK, errorMessage = pcall(function()
+						-- store values for error reporting
+						existingValue = ref.instance[key]
```

Comment:
I would imagine it's better to fetch this information when an error is encountered, as this prevents us from introducing more overhead when an error doesn't occur (which is far more often)

Reply:
Thanks for the feedback! I agree, I'll work on addressing that right now.

Reply:
I pushed a fix for this issue, but I'm leaving this comment unresolved so you can verify this fix properly address your comment.